pkg/frontend/querymiddleware/codec_test.go

```diff
@@ -112,6 +112,26 @@ func TestRequest(t *testing.T) {
 	}
 }
 
+func TestEncodeRequest_AcceptHeader(t *testing.T) {
+	for _, queryResultPayloadFormat := range allFormats {
+		t.Run(queryResultPayloadFormat, func(t *testing.T) {
+			codec := NewPrometheusCodec(prometheus.NewPedanticRegistry(), Config{QueryResultPayloadFormat: queryResultPayloadFormat})
+			req := PrometheusInstantQueryRequest{}
+			encodedRequest, err := codec.EncodeRequest(context.Background(), &req)
+			require.NoError(t, err)
+
+			switch queryResultPayloadFormat {
+			case formatJSON:
+				require.Equal(t, "application/json", encodedRequest.Header.Get("Accept"))
+			case formatProtobuf:
+				require.Equal(t, "application/vnd.mimir.queryresponse+protobuf,application/json", encodedRequest.Header.Get("Accept"))
+			default:
+				panic(fmt.Sprintf("unknown query result payload format: %v", queryResultPayloadFormat))
```

Comment:
Seems like this should be `t.Fatal(msg)`, no?

Reply:
Fixed in be89dde.